src/Request/Request.php

```diff
@@ -244,12 +244,20 @@ private function createRetryDecider(): callable
                 }
             }
 
+            if (is_object($response) && is_object($response->getBody()) && $response->getBody()->getContents() !== '') {
```

Comment:
This part looks okay-ish. A comment with an explanation would be helpful.

Reply:
i feel like checking if the thing we're about to reference is defined is kind of self-explanatory, but a comment saying
`// These values have been observed in the wild to be unset sometimes`